### ISSUE TITLE
how to exit program elegantly while set  -max_iters  to -1?

### ISSUE THREAD

**Issue by SeekPoint** [yukezhu/visual7w-qa-models#1]

**Comment by SeekPoint** [comment]
I believe if we just ctrl +c , it will cause a bad corrupted json file,

which in turn it will causes the error if we do it again:

question 380107: what color are the flowers ? red . 
question 380108: how is the weather ? rainy .  
evaluating validation performance... 28032 (7.895280)  
validation loss:    7.9089808119379 
validation accuracy:    0.2705122716895 
/home/rzai/torch/install/bin/luajit: ./misc/utils.lua:28: attempt to index local 'file' (a nil value)
stack traceback:
    ./misc/utils.lua:28: in function 'write_json'
    train_telling.lua:369: in main chunk
    [C]: in function 'dofile'
    ...rzai/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
    [C]: at 0x00406670
rzai@rzai00:~/prj/visual7w-qa-models$

**Comment by yukezhu** [comment]
It will be ok if you ctrl+c when it is passing through mini-batches, as no I/O is done during training. It will generate a corrupted json only when the program is trying to write its evaluation results to disk ([L369](https://github.com/yukezhu/visual7w-qa-models/blob/master/train_telling.lua#L369)). You can add code to prevent interruptions around that line.
